flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala

```diff
@@ -349,27 +351,29 @@ class JobManager(
 
               case JobStatus.CANCELED =>
                 jobInfo.client ! decorateMessage(
-                  Failure(
-                    new JobCancellationException(
-                      jobID,
-                    "Job was cancelled.", error)
+                  Failure(new SerializedThrowable(
+                      new JobCancellationException(
+                        jobID,
+                      "Job was cancelled.", deserializedError)
+                    )
                   )
                 )
 
               case JobStatus.FAILED =>
                 jobInfo.client ! decorateMessage(
-                  Failure(
-                    new JobExecutionException(
-                      jobID,
-                      "Job execution failed.",
-                      error)
+                  Failure(new SerializedThrowable(
+                      new JobExecutionException(
+                        jobID,
+                        "Job execution failed.",
+                        deserializedError)
+                    )
                   )
                 )
 
               case x =>
                 val exception = new JobExecutionException(jobID, s"$x is not a " +
                   "terminal state.")
-                jobInfo.client ! decorateMessage(Failure(exception))
+                jobInfo.client ! decorateMessage(Failure(new SerializedThrowable(exception)))
```

Comment:
Do we really need to serialize the `JobExecutionException` in a `SerializedThrowable` here?